tests/framework/static_analysis.py

```diff
@@ -0,0 +1,601 @@
+        )
+
+        self.instruction = insn
+
+
+class BackpropagationReachedStartOfFn(Exception):
+    """Exception indicating that the beginning of a function was reached during backpropagation, without any immediate
+    value being loaded into the register whose value we were trying to determine"""
+
+    def __init__(self, current_register):
+        super().__init__(
+            f"Backpropagation reached beginning of function definition while backpropagating {current_register}. Maybe it is a parameter itself?"
+        )
+
+
+def parse_objdump_output(output: str, arch: Architecture) -> list[Function]:
+    """Parse the stdout from obj dump into a list of the contained functions"""
+    lines = output.splitlines()
+
+    # Skip the first line of the output, it's just the file format
+    lines = lines[2:]
+
+    functions = []
+    current_function = None
+
+    for line in lines:
+        line = line.strip()
+
+        # Skip empty lines and those just announcing the start of a new section
+        if not line or line.startswith("Disassembly of section"):
+            # all functions are separated by empty lines. This is a sanity check to ensure the regex below
+            # catches all functions!
+            current_function = None
+            continue
+
+        # Start of a new function?
+        mtch = re.match(r"^<(.+)>:$", line)
+
+        if mtch:
+            # group 0 is always the full match (e.g. in our case the entire string because we have a regex with ^ and $)
+            # to get the groups defined inside the regex, start at 1.
+            current_function = Function(mtch.group(1), arch)
+            functions.append(current_function)
+
+            continue
+
+        # otherwise, must be instruction
+        if not current_function:
+            logger.error(
+                "Unexpectedly found data outside of function. Skipping line %s", line
+            )
+            continue
+
+        current_function.instructions.append(arch.instruction_from_str(line))
+
+    return functions
+
+
+def find_syscalls_in_binary(binary_path: Path):  # pylint: disable=too-many-branches
+    """Statically analyzes the given binary to find all syscalls.
+
+    Uses objdump's '-d' option, parses the output, and then at the call site of each syscall instruction
+    (and also of simple wrappers around it that weren't inlined during compilation), tries to determine the values
+    of registers holding arguments to the syscall instruction."""
+    if platform.processor() == "x86_64":
+        arch = X86_64()
+    else:
+        arch = Aarch64()
+
+    p_objdump = subprocess.Popen(
+        f"objdump -d {binary_path} --no-show-raw-insn --no-addresses".split(),
+        stdout=subprocess.PIPE,
+    )
+    # demangle rust symbols
+    disassembly = subprocess.check_output(["rustfilt"], stdin=p_objdump.stdout).decode(
+        "utf-8"
+    )
```

Comment:
`objdump -C rust` should do the trick as well.

Reply:
Neat! One less dependency to install, thanks